### ISSUE TITLE
Occurrence table should have a default sort order

### ISSUE THREAD

**Issue by zygoballus** [Symbiota/community-board#635]
### Symbiota Portal Name

Ecdysis

### Is your feature/tool request related to a problem? Please describe.

Since no sort order is specified by default when using `occurrencetabledisplay.php`, the order in which the results are displayed is formally unpredictable. In practice, the database will return them in the order they are stored in the database, but this can get jumbled, especially due to imports. Example screenshot:
<img width="365" alt="Screen Shot 2024-10-09 at 6 30 35 PM" src="https://github.com/user-attachments/assets/1e4c155f-71b0-439b-8901-aeb253c683c8">
If your occurrence table gets jumbled, the default table display can becomes mostly useless and you have to manually specify a sort order every time. To add insult to injury, there is no option in the interface to sort by Symbiota ID (`occid`), so it may not even be possible to get a table sorted by chronological insertion order (sorting by Date Entered doesn't work for imported records).

### Describe the solution you'd like.

By default, occurrencetabledisplay.php should sort by Symbiota ID (`occid`). Since this field is the primary key it shouldn't cause any performance issues. This will give a reliable and predictable table order by default.

This could probably be fixed by editing OccurrenceEditorManager->setSqlOrderBy().

**Comment by themerekat** [comment]
@zygoballus , is there a particular reason why you'd want to sort based on occid, rather than something like catalognumber or date last modified?

**Comment by zygoballus** [comment]
@themerekat - Yes, I think in most cases the most useful (and expected) sort order is the order in which the records were added to the collection, i.e. chronological order. Catalog Number is actually a string, so it doesn't give the sorting you would expect, e.g. CMNH-IZ-A1, CMNH-IZ-A10, CMNH-IZ-A100, CMNH-IZ-A1000, CMNH-IZ-A1001. Date last modified is sometimes useful, but I don't think it's a good default. Plus sorting on anything that isn't a key in the database is going to be potentially slow for large collections. Sorting by `occid` should also be the least disruptive, as it will probably not cause any change for collections that have been managed in a single portal from the start. It should just eliminate unexpected anomalies in the current default sorting.

**Comment by themerekat** [comment]
Roger. Performance is definitely my highest concern here.

**Comment by zygoballus** [comment]
Sorry for the bad initial pull request. I think this one should solve it correctly.

I've noticed this problem is more widespread than I thought, but I'm not really sure what causes it exactly (other than performing imports). Here's another example (from midatlanticherbaria.org) showing a jumbled default sort:

<img width="367" alt="Screen Shot 2024-11-11 at 12 02 46 PM" src="https://github.com/user-attachments/assets/4fafb3c5-91d2-4108-b281-128371496430">

**Comment by zygoballus** [comment]
Interestingly, this problem no longer occurs on Ecdysis. I wonder if the database there was reloaded or reindexed or something. The problem still occurs on midatlanticherbaria.org, though. The submitted patchset should keep it from happening regardless of the state of the database.

**Comment by themerekat** [comment]
The Mid-Atlantic Herbaria Consortium hasn't been updated to Symbiota 3.1, so it makes sense that you are seeing a discrepancy there.

**Comment by zygoballus** [comment]
@themerekat - Well, the weird thing is that the patchset to fix this hasn't been merged yet, so I wasn't expected it to be fixed. However, reloading the database (e.g. from a backup) or defragmenting it would also fix it.

**Comment by themerekat** [comment]
@zygoballus , right, I think the schema changes between 3.0 and 3.1 might be responsible for the change.

**Comment by themerekat** [comment]
@zygoballus , is this still an issue in 3.2?
